lib/files.js

```diff
@@ -0,0 +1,54 @@
+const { checkIfPathMatches } = require('./parseOwners')
+
+/**
+ * process the PR based on dependent settings
+ *
+ * @return
+ *  JSON object with the properties mergeable and the description if
+ *  `mergeable` is false
+ */
+module.exports = async (pr, context, settings) => {
+  let fileSettings = settings.files
+  if (!fileSettings) {
+    return {
+      mergeable: true,
+      description: 'Okay To Merge'
+    }
+  }
+
+  let patternToMatch = settings.files.pattern
+  // fetch the file list
+  let result = await context.github.pullRequests.getFiles(context.repo({number: pr.number}))
+  let addedFiles = result.data.filter(file => file.status === 'added')
```

Comment:
Any reason why you aren't including `modified` in here?

Reply:
no reason, fixed